com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JavaHotCodeReplaceProvider.java

```diff
@@ -343,4 +696,37 @@ public static IJavaElement findElement(String qualifiedTypeName, IJavaProject pr
         }
         return null;
     }
+
+    private boolean containsObsoleteMethods() throws DebugException {
+        List<ThreadReference> threads = currentDebugSession.getAllThreads();
+        for (ThreadReference thread : threads) {
+            if (!thread.isSuspended()) {
+                continue;
+            }
+            List<StackFrame> frames = getStackFrames(thread, false);
+            if (frames == null || frames.isEmpty()) {
+                continue;
+            }
+            for (StackFrame frame : frames) {
+                if (StackFrameUtility.isObsolete(frame)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private List<StackFrame> getStackFrames(ThreadReference thread, boolean refresh) {
```

Comment:
use compute/computeIfAbsent

Reply:
threadFrameMap.compute(thread, (key, old) -> {
            try {
                return old == null || refresh ? thread.frames() : old;
            } catch (IncompatibleThreadStateException e) {
                logger.log(Level.SEVERE, "Failed to get stack frames: " + e.getMessage(), e);
                return old;
            }
        });

Reply:
Fixed.